hosting/admin.py

```diff
@@ -92,7 +103,166 @@ def save_model(self, request, obj, form, change):
+  
+    # Creating VM by using method allocate(client, template)
+    def create_vm(self, request):
+        message = ''
+        # check if the request contains the template parameter, if it is 
+        # not set warn the user of setting this.
+        vm_template = request.POST.get('vm_template')
+        if vm_template == 'select' :
+            messages.add_message(request, messages.ERROR, "Please select a vm template")
+        else :
+            try :
+                # We do have the vm_template param set. Get and parse it
+                # and check it to be in the desired range.
+                vm_template_int = int(vm_template)
+                if vm_template_int >=1 and vm_template_int <= 8:
+                    # Lets create a test VM with 128MB of ram and 1 CPU
+                    vm_id = oca.VirtualMachine.allocate(self.client, '<VM><MEMORY>' + str(1024 * vm_template_int) + '</MEMORY><VCPU>' + str(vm_template_int)+ '</VCPU><CPU>' + str(0.1 * vm_template_int) + '</CPU><DISK><TYPE>fs</TYPE><SIZE>' + str(10000 * vm_template_int) + '</SIZE></DISK></VM>')
+                    message = "Created with id = " + str(vm_id)
+                    vm_pool = self.get_vms()
+                    messages.add_message(request, messages.SUCCESS, message)                    
+                else:
+                    messages.add_message(request, messages.ERROR, "Please select an appropriate value for vm template.")
+            except socket.timeout:
+                messages.add_message(request, messages.ERROR, "Socket timeout error.")
+            except OpenNebulaException:
+                messages.add_message(request, messages.ERROR, "OpenNebulaException occurred.")
+            except OSError as err:
+                messages.add_message(request, messages.ERROR, str("OS error: {0}".format(err)))      
+            except ValueError:
+                messages.add_message(request, messages.ERROR, "Please select an appropriate value for vm template.")
+        return redirect('admin:showvms')
+   
+    # Retrives virtual machine pool information
+    def get_vms(self):
+        vm_pool = oca.VirtualMachinePool(self.client)
+        vm_pool.info()
+        return vm_pool
+        
+    # Delete VM from the pool and DB by using method finalize()
+    def delete_vm(self, request, vmid):
+        # get the desired vm from the pool
+        vm_id = int(vmid)
+        vm = self.get_vm_by_id(vm_id)
+        if vm == -1:
+            messages.add_message(request, messages.ERROR, "Did not find a vm with id = " + str(vm_id))
+        else :
+            print("Deleting vm_id = " + str(vm_id) + " state = " + vm.str_state)
+            if vm.str_state == 'PENDING' or vm.str_state =='POWEROFF' or vm.str_state =='ACTIVE':
+                vm.delete()
+                messages.add_message(request, messages.SUCCESS, "Deleted from " + vm.str_state + " state vm with id = " + str(vm_id))
+            else:
+                vm.finalize()
```

Comment:
The method is defined in the orca VirtualMachine class, but I can't find any reference of it in the documentation [1], but if it's possible @telmich do we want to do this? 

The description for the delete method seems brute enough [2], and what are the reasons to avoid the "shutdown" or "shutdown --hard" actions?

1: http://docs.opennebula.org/4.10/integration/system_interfaces/api.html
2: http://docs.opennebula.org/4.10/user/virtual_resource_management/vm_guide_2.html